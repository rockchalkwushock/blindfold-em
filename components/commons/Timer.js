import SubTitle from './SubTitle'

const Timer = ({ activity, cooldown, timer }) => (
  <div>
    <SubTitle text={`Your activity: ${activity}`} />
    <SubTitle text={`Your timer: ${timer}`} />
    <SubTitle text={`Your cooldown: ${cooldown}`} />
  </div>
)

export default Timer
