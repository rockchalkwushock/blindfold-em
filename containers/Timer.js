import { Component } from 'react'
import moment from 'moment'
import PropTypes from 'prop-types'

import { Button } from '../components'

export const t = {
  STOPPED: 0,
  START: 1,
  PAUSED: 2
}

class Timer extends Component {
  static propTypes = {
    activity: PropTypes.string.isRequired,
    cooldown: PropTypes.string.isRequired,
    timer: PropTypes.string.isRequired
  }
  state = {
    baseTime: moment.duration(parseInt(this.props.timer, 10), 'minutes'),
    currentTime: moment.duration(parseInt(this.props.timer, 10), 'minutes'),
    cooldown: moment.duration(parseInt(this.props.cooldown, 10), 'minutes'),
    pomodoro: null,
    pomodoroState: t.STOPPED
  }
  pause = () => console.log('pause')
  reduceTime = () => {
    const currentTime = moment.duration(this.state.currentTime)
    currentTime.subtract(1, 'second')

    this.setState(state => ({ ...state, currentTime }))
  }
  start = () => {
    this.setState(state => ({
      ...state,
      pomodoro: setInterval(this.reduceTime, 1000),
      pomodoroState: t.START
    }))
  }
  stop = () => {
    if (this.state.pomodoro) {
      clearInterval(this.state.pomodoro)
    }

    this.setState(state => ({
      ...state,
      currentTime: moment.duration(this.state.baseTime),
      pomodoro: null,
      pomodoroState: t.STOPPED
    }))
  }
  render() {
    const { activity } = this.props
    const { baseTime, cooldown, currentTime } = this.state
    const base = `Selected Timer: ${baseTime.get('minutes')}:${baseTime.get(
      'seconds'
    )}`
    const current = `Current Time: ${currentTime.get(
      'minutes'
    )}:${currentTime.get('seconds')}`
    const cool = `Selected Cooldown: ${cooldown.get('minutes')}:${cooldown.get(
      'seconds'
    )}`
    return (
      <div>
        <h1>{activity}</h1>
        <h2>{base}</h2>
        <h2>{current}</h2>
        <h2>{cool}</h2>
        <Button
          className="start"
          condition={this.state.pomodoroState === 1}
          fn={this.start}
          text="Start"
        />
        <Button
          className="pause"
          condition={this.state.pomodoroState === 2}
          fn={this.pause}
          text="Pause"
        />
        <Button
          className="stop"
          condition={this.state.pomodoroState === 0}
          fn={this.stop}
          text="Stop"
        />
      </div>
    )
  }
}

export default Timer
