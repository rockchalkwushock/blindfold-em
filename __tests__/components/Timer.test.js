import React from 'react'
import { mount } from 'enzyme'

import Timer from '../../components/commons/Timer'

describe('Component: <Timer />', () => {
  test('should mount without exploding', () => {
    const tree = mount(<Timer activity="coding" cooldown="3" timer="12" />)
    expect(tree).toMatchSnapshot()
  })
})
