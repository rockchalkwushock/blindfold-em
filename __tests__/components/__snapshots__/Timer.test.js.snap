// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Timer /> should mount without exploding 1`] = `
<Timer
  activity="coding"
  cooldown="3"
  timer="12"
>
  <div>
    <SubTitle
      text="Your activity: coding"
    >
      <h2>
        Your activity: coding
      </h2>
    </SubTitle>
    <SubTitle
      text="Your timer: 12"
    >
      <h2>
        Your timer: 12
      </h2>
    </SubTitle>
    <SubTitle
      text="Your cooldown: 3"
    >
      <h2>
        Your cooldown: 3
      </h2>
    </SubTitle>
  </div>
</Timer>
`;
